test/crc32_test.c

```diff
@@ -0,0 +1,198 @@
+/* crc32_test.c -- unit test for crc32 in the zlib compression library
+ * Copyright (C) 2019 Rogerio Alves <rogealve@br.ibm.com>, IBM
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+#else
+#  include "zlib-ng.h"
+#endif
+
+typedef struct {
+    int line;
+    uLong crc;
+    char* buf;
+    int len;
+    uLong expect;
+} crc32_test;
```

Comment:
I actually ifdef'd this struct's definition to match the different crc32 declarations for the compat and zlib-ng APIs. When building the default API, it will use fixed int types.
This way we was use the proper argument types and avoid potential problems with "hidden" type casting.